### PR TITLE
PLANNER-2226 - Prune kogito dep tree + adjust assembly scripts to 11

### PR DESCRIPTION
**Pull request by ge0ffrey**

### CODE REVIEW

File: optaplanner-quarkus-integration/optaplanner-quarkus/runtime/pom.xml [hunk 1]
```diff
@@ -27,16 +27,19 @@
...
         </exclusion>
       </exclusions>
     </dependency>
     <dependency>
+      <!-- Static replaces dynamic -->
       <groupId>org.kie.kogito</groupId>
       <artifactId>drools-core-static</artifactId>
     </dependency>
     <dependency>
+      <!-- TODO remove kogito-ruleunits with fix https://github.com/kiegroup/kogito-runtimes/pull/871 -->
```
Comment (rsynek) [inline optaplanner-quarkus-integration/optaplanner-quarkus/runtime/pom.xml:42]: The PR has been already merged.

File: optaplanner-distribution/src/main/assembly/scripts/runExamples.sh [hunk 2]
```diff
@@ -3,15 +3,14 @@
 # Change directory to the directory of the script
 cd `dirname $0`
 
-# Most examples run (potentially slower) with max heap of 128 MB (so -Xmx128m), but 1 example's dataset requires 1.5 GB
-jvmOptions="-Xms256m -Xmx1536m -Dorg.optaplanner.examples.dataDir=sources/data/"
+# Most examples run (potentially slower) with max heap of 128 MB (so -Xmx128m), but 1 example's dataset requires 2 GB
+jvmOptions="-Xms256m -Xmx2g -Dorg.optaplanner.examples.dataDir=sources/data/"
```
[the remark this answers is not in the document]
Comment (ge0ffrey) [reply optaplanner-distribution/src/main/assembly/scripts/runExamples.sh:7]: In some garbage collectors, 1.5G is not enough for Machine Reassignment, or it's in the "gray zone" (enough but impacts perf badly a lot).
These days, 2g is far less of a concern than it was 10 years ago

File: optaplanner-quarkus-integration/optaplanner-quarkus/runtime/pom.xml (hunk 1 above)
Comment (ge0ffrey) [reply optaplanner-quarkus-integration/optaplanner-quarkus/runtime/pom.xml:42]: Ah nice, let me try that out.

**Review by triceo — APPROVED** [review]

**Comment by ge0ffrey** [comment]
Merging already to help Lukas, meanwhile looking at Radovan's suggestion

File: optaplanner-quarkus-integration/optaplanner-quarkus/runtime/pom.xml (hunk 1 above)
Comment (ge0ffrey) [reply optaplanner-quarkus-integration/optaplanner-quarkus/runtime/pom.xml:42]: https://github.com/kiegroup/optaplanner/pull/1017

**Pull request merged**